lib/rpush/daemon/fcm/delivery.rb

```diff
@@ -0,0 +1,160 @@
+module Rpush
+  module Daemon
+    module Fcm
+      # https://firebase.google.com/docs/cloud-messaging/server
+      class Delivery < Rpush::Daemon::Delivery
+        include MultiJsonHelper
+
+        HOST = 'https://fcm.googleapis.com'.freeze
+        SCOPE = 'https://www.googleapis.com/auth/firebase.messaging'.freeze
+
+        def initialize(app, http, notification, batch)
+          if necessary_data_exists?
+            @app = app
+            @http = http
+            @notification = notification
+            @batch = batch
+
+            @uri = URI.parse("#{HOST}/v1/projects/#{ENV['FIREBASE_PROJECT_ID']}/messages:send")
```

Comment:
I think it's better to add firebase_project_id and other firebase credentials to the app model instead of the env variables.
There are projects which are supporting multi-project functionality such as "WhiteLabel" apps. At this point it will be an issue to support multiple apps.
Why not add credentials to the Rpush::Fcm::App in the same way as Rpush::Apnsp8::App?
 
E.g: 
```
    app = Rpush::Fcm::App.new(
      name: 'My app',
      project_id: ENV['FIREBASE_PROJECT_ID'],
      account_type: ENV['FIREBASE_ACCOUNT_TYPE'],
      client_id: ENV['FIREBASE_CLIENT_ID'],
      client_email: ENV['FIREBASE_CLIENT_EMAIL'],
      private_key: ENV['FIREBASE_PRIVATE_KEY']
    )
    app.save!


then just get these credentials from the app.
```
@uri = URI.parse("#{HOST}/v1/projects/#{app.project_id}/messages:send")

Thanks!